### PR TITLE
Drag'n'drop target visuals

### PR DESCRIPTION
**Pull request by fiskus**
Screenshot from 2021-03-17 12-16-06

### CODE REVIEW

File: catalog/app/containers/Bucket/PackageDialog/PackageDialog.tsx [hunk 1]
```diff
@@ -551,7 +566,12 @@ export const MetaInput = React.forwardRef(function MetaInput(
         </Lab.ToggleButtonGroup>
       </div>
 
-      <div {...getRootProps({ className: classes.dropzone })} tabIndex={undefined}>
+      <div
+        {...getRootProps({
+          className: classes.dropzone,
+        })}
+        tabIndex={undefined}
+      >
```
Comment (nl0) [inline catalog/app/containers/Bucket/PackageDialog/PackageDialog.tsx:574]: this change doesnt seem necessary (or is it prettier?)

File: catalog/app/utils/dragging.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    document.addEventListener('dragover', () => {
+      setDragActive(true)
+    })
+    document.addEventListener('dragleave', () => {
+      setDragActive(false)
+    })
+    document.addEventListener('drop', () => {
+      setDragActive(false)
+    })
+  }, [])
```
Comment (nl0) [inline catalog/app/utils/dragging.ts:16]: probably we should remove event listeners in the cleanup function

**Review by nl0 — CHANGES_REQUESTED** [review]
overall looks good, some comments inline, plus not very nice looking gaps appeared around the file list:

borders

File: catalog/app/utils/dragging.ts [hunk 3]
```diff
@@ -3,17 +3,20 @@ import * as React from 'react'
 export default function useDragging() {
   const [isDragActive, setDragActive] = React.useState(false)
 
+  const setActive = React.useCallback(() => setDragActive(true), [])
+  const setInactive = React.useCallback(() => setDragActive(false), [])
```
Comment (nl0) [inline catalog/app/utils/dragging.ts:7]: doesnt eslint complain about non-exhaustive dependencies? i know that setState is stable and can be omitted from dep list, but i thought eslint doesnt know this

**Review by nl0 — CHANGES_REQUESTED** [review]
now files border is made invisible and the dashed outline is inset
files-border

also, meta outline pushes the contents when active, and the original border is also hidden, which doesnt look very nice imo, especially at the junctions:
meta-borders

what i propose is just overlaying the outline like this:
border-overlay

File: catalog/app/utils/dragging.ts (hunk 3 above)
Comment (fiskus) [reply catalog/app/utils/dragging.ts:7]: Eslint knows this

**Review by nl0 — APPROVED** [review]

**Pull request merged**